### PR TITLE
Overhaul MeshVisual

### PR DESCRIPTION
**Pull request by bollu**
Solved the bug [described here](https://github.com/vispy/vispy/pull/928). 

#### TODO:
- [ ] Blinn Phong for MeshVisual
- [ ] Textures for MeshVisual

### CODE REVIEW

File: examples/basics/visuals/mesh.py [hunk 1]
```diff
@@ -52,9 +52,9 @@ def __init__(self):
         vcolor[:, 1] = np.random.normal(size=nv)
         vcolor[:, 2] = np.linspace(0, 1, nv)
         self.meshes.append(visuals.MeshVisual(verts, faces, vcolor))
-        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor, 
-                                              shading='flat'))
-        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor, 
+        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor,
+                                              shading=' flat'))
```
Comment (campagnola) [inline examples/basics/visuals/mesh.py]: ?

File: examples/basics/visuals/mesh.py [hunk 2]
```diff
@@ -83,7 +83,8 @@ def on_draw(self, ev):
         gloo.set_viewport(0, 0, *self.physical_size)
         gloo.clear(color='black', depth=True)
         for mesh in self.meshes:
-            mesh.draw(mesh.tr_sys)
+            mesh._prepare_transforms(mesh)
```
Comment (campagnola) [inline examples/basics/visuals/mesh.py]: don't call `_prepare_transforms`; try copying the `on_resize`method from other examples.

File: examples/basics/visuals/mesh.py [hunk 3]
```diff
@@ -52,9 +52,9 @@ def __init__(self):
...
         self.meshes.append(visuals.MeshVisual(verts, faces, vcolor))
-        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor, 
-                                              shading='flat'))
-        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor, 
+        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor,
+                                              shading=' flat'))
+        self.meshes.append(visuals.MeshVisual(verts, faces, vcolor,
                                               shading='smooth'))
 
         # Lay out meshes in a grid
```
Comment (campagnola) [inline examples/basics/visuals/mesh.py]: You can remove `tr_sys` and `mesh.tr_sys` below--all visuals now come with their own `TransformSystem`.

File: examples/basics/visuals/mesh.py (hunk 1 above)
Comment (bollu) [inline examples/basics/visuals/mesh.py]: whoops. that's totally a bug on my part. Must have leaked in while editing

**Comment by bollu** [comment]
The number of coordinate systems have increased:
```
(visual | scene | document | canvas | framebuffer | render)
```

how do they all relate?

**Comment by campagnola** [comment]
See: https://github.com/vispy/vispy/pull/928/files#diff-e50a70d2c0fceb7f0b7e407f0b36c3f0L11

**Comment by bollu** [comment]
@campagnola I've also asked you a doubt on the main vispy thread (https://github.com/vispy/vispy/pull/928) about blinn-phong. It'd help if I could get an answer to the doubt!

**Comment by bollu** [comment]
This is dead, what was do be done was done in #10. Textures will come later because there's an API design discussion needed for that. So, I'm closing this.

**Pull request closed without merging**